internal/internal_workflow_testsuite.go

```diff
@@ -2358,6 +2358,17 @@ func (env *testWorkflowEnvironmentImpl) ExecuteNexusOperation(
 ) int64 {
 	seq := env.nextID()
 	taskHandler := env.newTestNexusTaskHandler()
+	// Use lower case header values to simulate how the Nexus SDK (used internally by the "real" server) would transmit
+	// these headers over the wire.
+	nexusHeader := make(map[string]string, len(params.nexusHeader))
+	for k, v := range params.nexusHeader {
+		nexusHeader[strings.ToLower(k)] = v
+	}
+	params.nexusHeader = nexusHeader
+	// The real server allows requests to take up to 10 seconds, mimic that behavior here.
+	// Note that if a user sets the Request-Timeout header, it gets overridden.
+	params.nexusHeader[strings.ToLower(nexus.HeaderRequestTimeout)] = "10s"
```

Comment:
Isn't the real server behaviour the minimum of 10s or the remaining schedule to close?

Reply:
Good point. Not at the moment, but we may enforce that eventually.